aiocache/base.py

```diff
@@ -498,12 +489,13 @@ async def close(self, *args, _conn=None, **kwargs):
     async def _close(self, *args, **kwargs):
         pass
 
-    def _build_key(self, key, namespace=None):
-        if namespace is not None:
-            return "{}{}".format(namespace, _ensure_key(key))
-        if self.namespace is not None:
-            return "{}{}".format(self.namespace, _ensure_key(key))
-        return key
+    def build_key(self, key, namespace=None):
+        key_name = key.value if isinstance(key, Enum) else key
+        ns = namespace if namespace is not None else self.namespace
+        return self._build_key(key_name, namespace=ns)
+
+    def _build_key_default(self, key, namespace=None):
```

Comment:
I'm still inclined to just pass a lambda as the default value in `__init__()`.

Reply:
Yeah, maybe you're right.  I had thought that this could provide a nice 'template' for end users who write their own key_builder; and it's clear when it gets overridden (like in the redis backend).

But if you think that the benefit is minimal, I'm ok with using the lambda.